packages/flutter_tools/lib/src/commands/build_macos_framework.dart

```diff
@@ -0,0 +1,305 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/process.dart';
+import '../base/utils.dart';
+import '../build_info.dart';
+import '../build_system/build_system.dart';
+import '../build_system/targets/macos.dart';
+import '../cache.dart';
+import '../flutter_plugins.dart';
+import '../globals.dart' as globals;
+import '../macos/cocoapod_utils.dart';
+import '../runner/flutter_command.dart' show DevelopmentArtifact, FlutterCommandResult;
+import '../version.dart';
+import 'build_ios_framework.dart';
+
+/// Produces a .framework for integration into a host macOS app. The .framework
+/// contains the Flutter engine and framework code as well as plugins. It can
+/// be integrated into plain Xcode projects without using or other package
+/// managers.
+class BuildMacOSFrameworkCommand extends BuildFrameworkCommand {
```

Comment:
I wanted to share more between this and `BuildIOSFrameworkCommand` but almost every line is subtly different.
For example, the iOS assemble target only creates App.framework, but the macOS one creates App.framework and copies FlutterMacOS.framework, so a separate copy isn't needed.  iOS publishes Flutter.xcframework, but macOS publishes FlutterMacOS.framework, which means the xcframework needs to be created.  iOS has extra logic for building simulator frameworks, etc.